WordPress/Classes/Services/SuggestionService.swift

```diff
@@ -0,0 +1,118 @@
+    @return An array of suggestions
+    */
+    @objc func suggestions(for siteID: NSNumber) -> [Suggestion]? {
+        if let cachedSuggestions = suggestionsCache.object(forKey: siteID) {
+            return cachedSuggestions.value
+        }
+        updateSuggestions(for: siteID)
+        return nil
+    }
+
+    /**
+    Performs a REST API request for the siteID given.
+
+    @param siteID ID of the blog/site to retrieve suggestions for
+    */
+    private func updateSuggestions(for siteID: NSNumber) {
+
+        // if there is already a request in place for this siteID, just wait
+        guard !siteIDsCurrentlyBeingRequested.contains(siteID) else { return }
+
+        // add this siteID to currently being requested list
+        siteIDsCurrentlyBeingRequested.append(siteID)
+
+        let suggestPath = "rest/v1.1/users/suggest"
+        let context = ContextManager.shared.mainContext
+        let accountService = AccountService(managedObjectContext: context)
+        let defaultAccount = accountService.defaultWordPressComAccount()
+        let params = ["site_id": siteID]
+
+        defaultAccount?.wordPressComRestApi.GET(suggestPath, parameters: params, success: { [weak self] responseObject, httpResponse in
+            guard let `self` = self else { return }
+            guard let payload = responseObject as? [String: Any] else { return }
+            guard let restSuggestions = payload["suggestions"] as? [[String: Any]] else { return }
+
+            let suggestions = restSuggestions.map { Suggestion(dictionary: $0) }
+            self.suggestionsCache.setObject(Wrapper(suggestions), forKey: siteID)
+
+            // send the siteID with the notification so it could be filtered out
+            NotificationCenter.default.post(name: .suggestionListUpdated, object: siteID)
+
+            // remove siteID from the currently being requested list
+            self.siteIDsCurrentlyBeingRequested.removeAll { $0 == siteID}
+        }, failure: { [weak self] error, _ in
+            guard let `self` = self else { return }
+
+            // remove siteID from the currently being requested list
+            self.siteIDsCurrentlyBeingRequested.removeAll { $0 == siteID}
+
+            DDLogVerbose("[Rest API] ! \(error.localizedDescription)")
+        })
+    }
+
+    /**
+    Tells the caller if it is a good idea to show suggestions right now for a given siteID.
+
+    @param siteID ID of the blog/site to check for
+    @return BOOL Whether the caller should show suggestions
+    */
+    @objc func shouldShowSuggestions(for siteID: NSNumber?) -> Bool {
+        guard let siteID = siteID, let appDelegate = WordPressAppDelegate.shared else {
+            return false
+        }
+
+        let suggestions = suggestionsCache.object(forKey: siteID)?.value
+
+        // if the device is offline and suggestion list is not yet retrieved
+        if !appDelegate.connectionAvailable && suggestions == nil {
```

Comment:
Rather than accessing the app delegate here, I think it would be better to use `ReachabilityUtils.isInternetReachable()` (does the same thing, but abstracted a layer)

Reply:
Thanks! I hadn't noticed that utility.